TACC-Cloud/hazmapper
# task/WG-251 Add Tile Server Hook

## Overview: ##

Added hook to retrieve tile servers and added its usage to the map

## PR Status: ##


## Related Jira tickets: ##

* [WG-251](https://tacc-main.atlassian.net/browse/WG-251)

## Summary of Changes: ##

1. Added hook to get tile servers from the geoapi endpoint 
2. Added hook usage to MapProjects to get tile servers of a project

## Testing Steps: ##

1. Switch react app to use staging environment by modifying `secret_local.ts` to use `GeoapiBackendEnvironment.Staging`
2. Go to [https://hazmapper.tacc.utexas.edu/staging/](https://hazmapper.tacc.utexas.edu/staging/) and open any existing map and copy the maps UUID from the url 
3. Go to `localhost:4200/project/{UUID}` and ensure that tile servers are retrieved correctly. Can monitor the network tab in both staging and local environment to ensure that response is the same
4. Additionally make changes to the layers on the staging map and ensure changes are reflected correctly on local

## UI Photos:

No UI changes

## Notes: ##
